File: 프로그래머스/2/131130. 혼자 놀기의 달인/혼자 놀기의 달인.js
function solution(cards) {
    cards = cards.map(v => v - 1)

    let group = cards.map((v,i) => { 
        let save = 0
        return Array.from(new Set(cards.map(a => {
            save = cards[v]
            v = save
            return save
        }))).length
    })

    const result = Array.from(new Set(group.sort((a,b) => b - a)))
    if (result[0] === cards.length) return 0
    else if (result.length === 1) return result[0] * result[0]
    else return result[0] * result[1]
}

File: 프로그래머스/2/131130. 혼자 놀기의 달인/혼자 놀기의 달인.py
def solution(cards):
    answer = []
    
    for i in range(len(cards)):
        group = 0
        while cards[i]:
            ni = cards[i] - 1
            cards[i], i = 0, ni
            group += 1
        answer.append(group)
        
    answer.sort()
    return answer[-1] * answer[-2]